File: backend/app/models/cycle.py
from datetime import datetime, date
from uuid import UUID
from sqlalchemy import String, ForeignKey, Boolean, Integer, Date, DateTime
from sqlalchemy.orm import Mapped, mapped_column, relationship
from .base import Base, AuditMixin, SoftDeleteMixin

class Cycle(Base, AuditMixin, SoftDeleteMixin):
    __tablename__ = "cycles"
    
    name: Mapped[str] = mapped_column(String(255), nullable=False)
    description: Mapped[str | None] = mapped_column(String(1000), nullable=True)
    
    start_date: Mapped[date] = mapped_column(Date, nullable=False)
    end_date: Mapped[date | None] = mapped_column(Date, nullable=True)
    completed_at: Mapped[datetime | None] = mapped_column(DateTime, nullable=True)
    
    project_id: Mapped[UUID] = mapped_column(ForeignKey("projects.id"), nullable=False)
    workspace_id: Mapped[UUID] = mapped_column(ForeignKey("workspaces.id"), nullable=False)
    
    project: Mapped["Project"] = relationship(back_populates="cycles")
    workspace: Mapped["Workspace"] = relationship()
    issue_links: Mapped[list["IssueCycle"]] = relationship(back_populates="cycle")

File: backend-go/internal/model/cycle.go
package model

import "time"

// Cycle represents an iteration/sprint cycle.
type Cycle struct {
	BaseModel

	Name        string     `gorm:"size:255;not null" json:"name"`
	Description *string    `gorm:"size:1000" json:"description"`
	StartDate   time.Time  `gorm:"type:date;not null" json:"start_date"`
	EndDate     *time.Time `gorm:"type:date" json:"end_date"`
	CompletedAt *time.Time `json:"completed_at"`
	CancelledAt *time.Time `json:"cancelled_at"`
	ProjectID   uint64     `gorm:"not null;index" json:"project_id"`
	WorkspaceID uint64     `gorm:"not null" json:"workspace_id"`

	// Relationships
	Project    Project      `gorm:"foreignKey:ProjectID" json:"-"`
	IssueLinks []IssueCycle `gorm:"foreignKey:CycleID" json:"-"`
}

func (Cycle) TableName() string {
	return "cycles"
}
